http-auth/routes/routes_test.go

```diff
@@ -0,0 +1,179 @@
+package routes
+
+import (
+	"html"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"golang.org/x/time/rate"
+)
+
+func TestMain(t *testing.T) {
+	t.Run("GET /", func(t *testing.T) {
+		request, _ := http.NewRequest(http.MethodGet, "/", nil)
+		rr := httptest.NewRecorder()
+
+		IndexHandler(rr, request)
+
+		AssertStatusOK(t, rr)
+	})
+
+	t.Run("POST /", func(t *testing.T) {
+		bodyHTML := "<em>Hello Test</em>"
+		b := strings.NewReader(bodyHTML)
+		request, _ := http.NewRequest(http.MethodPost, "/", b)
+		request.Header.Add("Content-Type", "text/html")
+
+		rr := httptest.NewRecorder()
+
+		IndexHandler(rr, request)
+
+		AssertStatusOK(t, rr)
+
+		AssertBodyEquals(t, rr, HTMLHeader+html.EscapeString(bodyHTML))
```

Comment:
I would typically hard-code the exact string you expect to be returned here, rather than computing it via `html.EscapeString` - it's really easy to spot when a string literal isn't what you expect, and really hard to notice a bug when a computed value isn't computed correctly.

Test code has slightly different trade-offs than production code - one of the primary goals of test code is being _obviously_ correct at a glance (because we don't tend to write tests for tests!)

Reply:
Makes sense, thanks!